### ISSUE TITLE
sc2: SC2-compatible NeuralDQN with available-actions masking

### ISSUE THREAD

**Issue by espenhk** [espenhk/gamer-ai#106]
The existing `NeuralDQNPolicy` (`framework/policies.py`) implements deep Q-learning with an experience replay buffer and target network. It works with SC2's 9-cell discrete grid but ignores PySC2's `available_actions` mask, meaning the Q-network may assign high Q-values to functions that cannot legally execute (e.g. `Move_screen` before `select_army` is issued). The client silently falls back to `no_op` in these cases, producing meaningless gradient updates.

### 1. Pass `available_actions` through the info dict

In `games/sc2/client.py` `_timestep_to_obs_info()`, add:
```python
info["available_fn_ids"] = set(obs["available_actions"])
```

In `games/sc2/actions.py`, expose the mapping from discrete grid index → `fn_id` so the policy can construct a boolean mask over the 9-cell action grid.

### 2. Q-value masking before action selection

```python
# In SC2NeuralDQNPolicy.act():
q_values = self._q_network.forward(obs)            # shape (9,)
mask = self._build_mask(info["available_fn_ids"])  # shape (9,) bool
q_values[~mask] = -np.inf                          # mask illegal actions
action_idx = np.argmax(q_values)
```

During training, apply the same mask in both online and target network outputs to prevent gradient flow through illegal Q-values.

### 3. Episode length and replay buffer tuning

SC2 ladder episodes run up to 120 s (vs 30 s for TMNF), generating ~5× more transitions per episode. Recommended SC2 defaults:

| Param | SC2 default | TMNF default | Reason |
|---|---|---|---|
| `replay_buffer_size` | `50000` | `10000` | More transitions per episode |
| `min_replay_size` | `2000` | `500` | Proportional warm-up |
| `epsilon_decay_steps` | `20000` | `5000` | Longer exploration phase |
| `gamma` | `0.995` | `0.99` | Sparse win/loss reward needs long-horizon credit |

## Tasks

- [ ] Extend `SC2Client._timestep_to_obs_info()` to include `available_fn_ids` in the info dict (`games/sc2/client.py`)
- [ ] Add `discrete_action_to_fn_id()` lookup in `games/sc2/actions.py` (maps 9-cell grid index → PySC2 function ID)
- [ ] Add `_build_available_actions_mask(available_fn_ids, n_actions)` helper
- [ ] Implement `SC2NeuralDQNPolicy` (subclass of `NeuralDQNPolicy`) with masking applied in `act()` and `update()`
- [ ] Add SC2-recommended hyperparams to `games/sc2/config/training_params.yaml` under `policy_params`
- [ ] Register as `sc2_neural_dqn` in the policy factory (`framework/training.py` / `main.py`)
- [ ] Tests: masked Q-values → illegal action never chosen; replay buffer transitions contain correct mask; gradient does not flow through masked logits

## Relevant files

- `framework/policies.py` — `NeuralDQNPolicy` base class
- `games/sc2/client.py` — observation extraction, `_timestep_to_obs_info()`
- `games/sc2/actions.py` — `DISCRETE_ACTIONS`, grid index → fn_id mapping
- `games/sc2/config/training_params.yaml` — hyperparams
- New: `games/sc2/sc2_policies.py` — SC2-specific policy implementations
